diff --git a/bin/uefi-firmware-parser b/bin/uefi-firmware-parser
index 17d8668..7995959 100755
--- a/bin/uefi-firmware-parser
+++ b/bin/uefi-firmware-parser
@@ -22,8 +22,8 @@ def _process_show_extract(parsed_object):
         autodir = "%s_output" % FILENAME
         if os.path.exists(autodir):
             print("Skipping %s (_output directory exists)..." % (FILENAME))
-            return
-        os.makedirs(autodir)
+        else:
+            os.makedirs(autodir)
         args.output = autodir
 
     if args.extract:
